pkg/apps/strategy/recreate/recreate.go

```diff
@@ -234,6 +234,10 @@ func (s *RecreateDeploymentStrategy) scaleAndWait(deployment *kapi.ReplicationCo
 		if scaleErr == nil {
 			return true, nil
 		}
+		// Handle conflicts (shouldn't this be already handled by the scaler?)
```

Comment:
it was bugged.  https://github.com/kubernetes/kubernetes/pull/61790 comes to mind.

Reply:
That only fixes the result, not the cause. Do we want the scalers to handle conflicts? Personally, the current approach is fine, it's the callers duty to handle those.

Reply:
This is another reason we should not use kubectl scaler but rather directly implement that functionality.